java-util/src/main/java/io/druid/java/util/common/concurrent/Execs.java

```diff
@@ -147,4 +151,32 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor)
         }
     );
   }
+
+  private static final AtomicLong fjpWorkerThreadCount = new AtomicLong(0L);
+
+  public static ForkJoinWorkerThread makeWorkerThread(String name, ForkJoinPool pool)
+  {
+    final ForkJoinWorkerThread t = new ForkJoinWorkerThread(pool)
+    {
+      // No special handling in subclass
+    };
+    t.setDaemon(true);
+    final long threadNumber = fjpWorkerThreadCount.incrementAndGet();
+    t.setName(StringUtils.nonStrictFormat(name, threadNumber));
```

Comment:
There is no guarantees that `checkThreadNameFormat` must be called first.

Since the thread factories are called from some potentially asynchronous mechanism, I figure having the safest possible naming enforcement here is the most defensive